pkg/overlay/cache_test.go

```diff
@@ -141,3 +144,46 @@ func testCache(ctx context.Context, t *testing.T, store overlay.DB, sdb statdb.D
 		assert.True(t, err == overlay.ErrEmptyNode)
 	}
 }
+
+func TestRandomizedSelection(t *testing.T) {
+	t.Parallel()
+
+	totalNodes := 10
+	selectIterations := 100
+	numNodesToSelect := 1
+	minSelectCount := (selectIterations * numNodesToSelect / totalNodes) / 2
+
+	testplanet.Run(t, testplanet.Config{
```

Comment:
Use here `satellitedbtest.Run`, we want to setup at least 1000 storage nodes in the database.